Update Compromised TAM threat description

mingpeiwk opened the pull request:
Ben Schwartz comment on Section 9. Issue #251

mingpeiwk committed:
Address hostile TAM threats

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1347,6 +1347,16 @@ still a threat present in allowing a potentially compromised TAM to install
 malicious TA.
 
+There are also threats of hostile or abusive TAMs where a TAM turns to act out of 
+expectation of device administrators, for example, pushing out TAs that contain 
```
```suggestion
expectation of Device Administrators, for example, pushing out TAs that contain 
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1347,6 +1347,16 @@ still a threat present in allowing a potentially compromised TAM to install
+The mitigation methods for a compromised TAM case above can also apply to these threats.
+
+A Device Administrator may need to be notified when such an incident takes place.
+An Untrusted Application or some software component in the REE may monitor detection
+output from the device's TEEs, and notify the Device Administrator for additional 
+remediations.
```
I find this sentence confusing.  What is "detection output"?   And it's not clear to me that there is any such thing in general, so I'd rather omit this sentence and maybe this whole paragraph.

mingpeiwk replied:
This is a possibility similar to the suggested possible mitigation methods 1) - 3) above, which is out-of-scope for this doc to give actual specifications. A notification scheme is out of scope for this doc to notify an administrator. This is purely to mention that such a support should be considered as part of threat mitigation.

Let me remove this for now, and follow up later if anyone raises a request. Thank you for your suggestion here.

mingpeiwk committed:
Update draft-ietf-teep-architecture.md

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Revise the notification support statement

Change to not give a specific notification method because it can vary and it is out of scope. Just add a sentence that says some notification support should be provided for Device Administrators to be aware of possible incidents from a compromised or hostile TAM to their devices.

mingpeiwk commented:
Revised to just mention the need to support "Device Administrator notification" as asked in Ben Schwarts's comment.

mingpeiwk commented:
@dthaler please review again, and see if this is good to proceed.

bremoran commented:
There are other available mitigation methods for a hostile TAM. I should point out that a hostile TAM (working as intended, under control of intended parties, but malicious) is different from a compromised TAM (not working as intended, or under control of unintended parties, malicious). This has implications for how we mitigate, so the mitigations in Compromised TAM don't apply quite the same way.

Mitigations:
1. ACLs: Make explicit which TAs a TAM is permitted to install. This limits the scope for damage.
2. Transparency Logs: A hostile TAM can be named and shamed.
3. TAM certificate revocation (probably as a result of 2)
4. TAM Removal by the device administrator (as a result of 1, 2, or some out-of-band detection technique)

What does not work is the out-of-band signed release log described in the Compromised TAM section. This assumes that the TAM's infrastructure is compromised, but that their signing infrastructure is not. A hostile TAM is "compromised" at the social level, so that technological control isn't helpful.

mingpeiwk commented:
> 

Agreed on the out-of-band signed release log effectiveness for this use case. A hostile TAM may make them to look consistent while a compromised TAM may not. I will split this into a separate threat as the content is larger than I hoped.

mingpeiwk committed:
Split hostile TAM into a separate threat

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1347,6 +1347,33 @@ still a threat present in allowing a potentially compromised TAM to install
 any TA it wants; for example, a known-compromised TA or an explicitly
 malicious TA.
 
+The device manufacturer may want to include a notification support that can allow
+Device Administrators be notified for incidents upon detection with some mitigation 
+methods as suggested above.
```
A compromised TAM can in theory compromise a TEE, and once you compromise a TEE, you may be able to prevent any such notifications from occurring by that device.  And outside the device, the device manufacturer may not be the right entity to provide things. So I don't understand how this sentence is actionable at present.

mingpeiwk replied:
I am thinking the common case where a TEE is embedded by a device manufacturer, and multiple TAMs, by some initial set of Trust Anchors. A compromised TAM can only manage TAs, not be able to compromise a TEE, in my view. It may attack a TEE environment but DoS - install many TAs etc but cannot corrupt TEE internals. A notification scheme could be, for example, some APIs that the TEE provides, allowing some others to build a notification app inside REE to get detection output in TEE. Both the detection and such monitoring APIs of any detection of incidents are out of scope of TEEP.

There may be other ways to get detected results when the detection is carried out inside the TEE, as we recommended use of method 1 with an ACL.

dthaler replied:
> I am thinking the common case where a TEE is embedded by a device manufacturer, and multiple TAMs, by some initial set of Trust Anchors.

I am not sure that is a common case for TEEP.

> A compromised TAM can only manage TAs, not be able to compromise a TEE, in my view.

Disagree.  Maybe in a particular implementation, but I'm not sure that would be common.
A TAM has to be able to update dependencies, like OP-TEE, TF-A, etc.  Any Trusted Component.

> It may attack a TEE environment but DoS - install many TAs etc but cannot corrupt TEE internals.

"Cannot" is just an example, we should not make any claim as to whether in general one can or cannot, as we cannot say what will be "common" with TEEP yet, and it sounds like we have different expectations on what will be common.

> A notification scheme could be, for example, some APIs that the TEE provides, allowing some others to build a notification app inside REE to get detection output in TEE.

I don't know what "detection output in TEE" means.  Any detection in the REE cannot have visibility into things in the TEE or the TEE would fail the definition of a TEE in terms of code confidentiality or data confidentiality.

> Both the detection and such monitoring APIs of any detection of incidents are out of scope of TEEP.

Agree, but I'm also not convinced it's actually feasible or worth noting in this architecture document that already passed WGLC, IETF LC, and IESG ballot.

> There may be other ways to get detected results when the detection is carried out inside the TEE, as we recommended use of method 1 with an ACL.

I disagree we should be adding any recommendation at this stage as that would reset all the last call I think, and potentially take a while to get to consensus on.

mingpeiwk replied:
Removed as this is controversial, and it can be considered out of scope. Monitoring of TAM and TA activities is usually out of scope for a protocol design spec.

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1347,6 +1347,33 @@ still a threat present in allowing a potentially compromised TAM to install
+
+## Hostile TAM
+
+There are also threats of hostile or abusive TAMs where a TAM turns to act out of 
```
```suggestion
There are also threats of hostile or abusive TAMs where a TAM is compromised and thus turns to act out of 
```
Is this what you mean by "turns"?  If instead you mean it's intentional by the TAM but not by the Device Administrator who chose to trust that TAM then

```suggestion
There are also threats of hostile or abusive TAMs where a TAM acts out of 
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1347,6 +1347,33 @@ still a threat present in allowing a potentially compromised TAM to install
+detecting and mitigating a hostile TAM:
+
+1. ACLs: Apply an ACL to TAMs, making explicit which TAs the TAM is permitted to install. 
+This limits the scope of a hostile TAM's impact.
```
I'm not sure it really limits it.  If you have a way of managing such an ACL by knowing things like the hash of the TA, then you really are acting like a TAM.  Besides, as shown in the "Keys and Certificate Types" section, it's the Code Signing keys that control what TA's can be installed, not the TAM keys per se.   In other words, no point in allowing installation of code that fails the code signing check.   (As opposed to personalization data only components, for which the text says "Note that Personalization Data is not included in the table above.")

mingpeiwk replied:
It is hard to implement, I agree. We already want to use Trust Anchor as the guard. Individual abusive TA can be removed but nothing prevents a TAM to reinstall it :) So the TAM needs to be blocked if it repeats, where the other mitigation methods #2,3,4 will do.

Let me remove this for now. The same should apply to the "Compromised TAM" section where ACLs is possible but not likely be effective.

dthaler replied:
My opinion: a hostile TAM vs a compromised TAM is the same case.   The Device Administrator is responsible for choosing what TAM to trust with their device.   Once they trust one, it going bad is just like Device Administrator's employee going bad.  You use social/contractual enforcement to deal with it, it's not a technical issue.

mingpeiwk replied:
I have removed ACL. But still keep them separate. The "Apply ACL in TAM" is a mitigation for Compromised TAM by a TAM provider itself. Do you feel strongly that we merge the two back to one but leave the paragraph content the same under the title "Compromised TAM" or "Compromised or Hostile TAM"?

Hostile is an active behavior similar to privacy invasion by ads companies.

dthaler replied:
A TEEP Agent and Device Administrator cannot distinguish (in any technical way) between a "hostile" TAM and a "compromised" TAM and so have to treat them as the same case.   The Device Administrator picks the TAM(s) and implicitly trusts them.   If there are multiple, I do think it's ok to let them manage different things (e.g., trust anchor store, TEEP agent code, secure firmware/OS, etc.) so ACL is ok if expressed appropriately in text.

bremoran replied:
The idea behind my original "apply an ACL" comment was this. Imagine a scenario like installing signed software. You may have both Vendor A and Vendor B in your root store, but Vendor B should not be able to sign Vendor A's software and Vendor A should not be able to sign Vendor B's software.

Similarly in TA space, a TAM may be trusted to manage some content but not other content. I can easily imagine a scenario where two large companies both insist on operating their own TAMs. They could include banks or technology companies, etc. Each of these TAMs then should have strict limits placed on *which TAs* they are allowed to install. Now, when I say "which TAs," it might be that a better abstraction is "which trust domains they may install TAs into."

dthaler replied:
Well in particular you could in theory say that if you have multiple TAMs, that only the TAM that installed a given trusted component can update it.   However, I think that would unnecessarily mix two different things from [Figure 4](https://www.ietf.org/archive/id/draft-ietf-teep-architecture-18.html#name-keys-and-certificate-types).  Trusted Component install is based on code signing keys not TAM keys.   So to install anything, you need a manifest signed by a code signing key in the TEE's code signing trust anchor store.   That store is separate from the TEEP Agent's store of TAM keys, as the figure shows.

For installation of _new_ trusted components, I think Brendan is right about "which trust domains they may install TAs into".   However per issues #7 and #62 the WG consensus was to remove all mention of security domains from the base architecture.  Similarly in SUIT, trust domains were separated out into an extension.   So it's difficult to reference them now, and I'd prefer some wording that does not pull that concept back in now, but ok in a separate document.

bremoran replied:
While my original wording was a little fuzzy, given that I'm not specifying **how** an ACL can specify which TAs a TAM is permitted to install, it does deal with the question of limiting damage without reintroducing trust domains. I think we should probably stick with that, leaving exactly what the ACL does up to implementors How about this:

ACLs: Make explicit which TAs a TAM is permitted to install. This limits the scope for damage. ACLs could restrict which path prefixes a TAM can use, which TA Components they are permitted to access, which TA Signers must have signed the TAs they install, etc.

mingpeiwk replied:
Reading Ben's comment again, I think I might have added more than it was asked, and therefore bring up controversy. 

Ben asked the following:

"1) What damage can a hostile TAM do? 
  2) What does the device administrator need to know for adding a trust anchor to be safe?"

I will revise to just focus on answering these two. The first one has largely answered in the first paragraph. There may be more damages to enumerate but the main ones are captured.

Then it is to respond the second point: we just say an admin has privilege to add or remove trust anchor to limit a TAM's management to a device. Other schemes are out of scope while possible.

Thoughts?

mingpeiwk replied:
Add that a compromised TAM cannot install arbitrary TA unless such a TA is signed by a trusted signer first. I have adjusted the threat description to reflect this.

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1347,6 +1347,33 @@ still a threat present in allowing a potentially compromised TAM to install
+This limits the scope of a hostile TAM's impact.
+2. Trust Anchor update: when a Trust Anchor entry in the TEE Trust Anchor list is specific to 
+the hostile TAM, a Trust Anchor update process may be used to remove the Trust Anchor as a way 
+to stop the TAM for installing TAs to the TEE.
```
```suggestion
to stop the TAM from installing TAs to the TEE.
```
But the doc says "Trust Anchors can be updated similarly to the Personalization Data for any other TA."
so a hostile TAM might modify the trust anchor set to remove any other trust anchors from being able to modify the trust anchor set.

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1347,6 +1347,33 @@ still a threat present in allowing a potentially compromised TAM to install
+the hostile TAM, a Trust Anchor update process may be used to remove the Trust Anchor as a way 
+to stop the TAM for installing TAs to the TEE.
+3. TAM certificate revocation: the certificate issuing authority of the TAM certificate is 
+requested to revoke the TAM's certificate. The TEEP Agents that supports certificate revocation 
```
```suggestion
requested to revoke the TAM's certificate. TEEP Agents that support a certificate revocation 
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1347,6 +1347,33 @@ still a threat present in allowing a potentially compromised TAM to install
+requested to revoke the TAM's certificate. The TEEP Agents that supports certificate revocation 
+status check will be able to stop the TAM, and further decide to remove those hostile TAs from that TAM.
+4. TA removal by the Device Administrator upon detection of a hostile TAM and any abusive TA.
+The detection cab be the result of 1) above, and some other out-of-band detection technique.
```
```suggestion
The detection can be the result of 1) above, and some other out-of-band detection technique.
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1347,6 +1347,33 @@ still a threat present in allowing a potentially compromised TAM to install
+4. TA removal by the Device Administrator upon detection of a hostile TAM and any abusive TA.
+The detection cab be the result of 1) above, and some other out-of-band detection technique.
+
+The device manufacturer may want to include a notification support that can allow
+Device Administrators be notified for incidents upon detection with some mitigation 
+methods as suggested above.
```
Seems like a duplicate of line 1350?

mingpeiwk replied:
Yes, I added this to mean notification should be supported for detection of a hostile TAM too. Do you prefer we better leave it out and someway the notification suggestion in the other Threat item is sufficient to cover?

dthaler replied:
Since we already went through last call, I would avoid any controversial topics if at all possible.   This is one.

mingpeiwk replied:
Brendon expanded the "Compromised TAM" section a lot. Shall we pull out some?

The "hostile TAM" comment is from Ben Schwarts, asking for notification to an admin. So this tries to address it. Notification is kind of out of scope, which means monitoring capability first before alert. Monitoring of a TAM and TEE is much extra.

Let me remove notification sentences in both. How about mitigations that Brendon added and my revised list above? Please review again. Thanks.

mingpeiwk committed:
Update draft-ietf-teep-architecture.md

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Update draft-ietf-teep-architecture.md

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Update draft-ietf-teep-architecture.md

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Update draft-ietf-teep-architecture.md

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Removed ACL approach for hostile TAM

An ACL may be applied at TAM for a TAM to control possible compromising but not useful for a hostile TAM. ACL isn't effective to be added in the device side where Trust Anchor is the protocol's authorization method against a TAM.

mingpeiwk committed:
Remove notification suggestions

Notification relies on monitoring first. Monitoring is considered out of scope for TEEP protocol, and architecture.

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1347,6 +1347,23 @@ still a threat present in allowing a potentially compromised TAM to install
+There are also threats of hostile or abusive TAMs where a TAM is compromised and thus turns to act out of 
+expectation of Device Administrators, for example, pushing out TAs that contain 
+some data collection or use users' device resources for distributed jobs for a TAM. 
+To mitigate this threat, TEEs and Device Administrators have several options for 
+detecting and mitigating a hostile TAM:
+
+1. Trust Anchor update: when a Trust Anchor entry in the TEE Trust Anchor list is specific to 
+the hostile TAM, a Trust Anchor update process may be used to remove the Trust Anchor as a way 
+to stop the TAM from installing TAs to the TEE.
+2. TAM certificate revocation: the certificate issuing authority of the TAM certificate is 
+requested to revoke the TAM's certificate. TEEP Agents that support a certificate revocation 
+status check will be able to stop the TAM, and further decide to remove those hostile TAs from that TAM.
+3. TA removal by the Device Administrator upon detection of a hostile TAM and any abusive TA.
+The detection can be the result of 1) above, and some other out-of-band detection technique.
```
My opinion: lines 1339-1343 and lines 1358-1365 require confirming consensus if left in, and have some risk of objections since they are not simply editorial "things an admin should think about"

Lines 1336-1337 can even be (mis?)read as implying the list is a complete list and there cannot be other options.

mingpeiwk replied:
I see your point ""things an admin should think about"" - Trust Anchor update may not be familiar to an ordinary Device Administrator which can be a device owner for the knowledge gap.

Device Administrators may contact "some" organization to stop or remove a TAM that turns to be bad, or contact the TAM to just stop it which can work in some cases, as to how we handle real world scenarios with social networking companies.

mingpeiwk replied:
@dthaler please review the updated paragraphs.

mingpeiwk committed:
Revise compromised and hostile TAM threat sections

Updated compromised TAM threat section: a TAM cannot install arbitrary TAs. There is the second Trust Anchors in the TEE that validates TA signer to be trust worthy or not. Also mentioned the risk in TAM side by leaking data to an adverse actor.

In the case of a hostile TAM, the mitigation focus changes to stop using the TAM, and switch to use another one.

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
-operator. To mitigate this threat, TEEs have several options for 
-detecting and mitigating a compromised TAM:
+to determine that the TAM is trustworthy. A compromised TAM may bring multiple threats 
+and damage to user devices that it can managee and the device owners. 
```
```suggestion
and damage to user devices that it can manage and thus to the Device Owners. 
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
-detecting and mitigating a compromised TAM:
+to determine that the TAM is trustworthy. A compromised TAM may bring multiple threats 
+and damage to user devices that it can managee and the device owners. 
+In one hand, the device information of many devices that the TAM manages and 
```
```suggestion
Information on devices that the TAM manages 
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
+to determine that the TAM is trustworthy. A compromised TAM may bring multiple threats 
+and damage to user devices that it can managee and the device owners. 
+In one hand, the device information of many devices that the TAM manages and 
+therefore have in its system may be leaked to a bad actor. On the other hand,
```
```suggestion
may be leaked to a bad actor.
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
+and damage to user devices that it can managee and the device owners. 
+In one hand, the device information of many devices that the TAM manages and 
+therefore have in its system may be leaked to a bad actor. On the other hand,
+a compromised TAM can install many TAs to launch somee DoS attack to devices,
```
```suggestion
A compromised TAM can also install many TAs to launch a DoS attack on devices,
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
+In one hand, the device information of many devices that the TAM manages and 
+therefore have in its system may be leaked to a bad actor. On the other hand,
+a compromised TAM can install many TAs to launch somee DoS attack to devices,
+for example, by filling up a device's TEE resources reserved for TAs such that 
+other TAs may not get resource to be installed or properly function. It may
```
```suggestion
for example, by filling up a device's TEE resources reserved for TAs such that 
other TAs may not get resources to be installed or properly function. It may
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
+a compromised TAM can install many TAs to launch somee DoS attack to devices,
+for example, by filling up a device's TEE resources reserved for TAs such that 
+other TAs may not get resource to be installed or properly function. It may
+also install malicious TAs to, potentially, many devices, if it can also 
+get or access a Trusted Component signer key that is trusted by the TEEs. 
```
This sentence is about a compromised Trusted Component Signer, not a compromised TAM, I think.  And so it doesn't really belong here in my view.

mingpeiwk replied:
It describes that unless the TAM can have a signer key that is trusted it isn't able to install a malicious or abusive TA. This is to describe a condition for this TAM's distribution of abusive TA. So I think it applies here.

dthaler replied:
I don't follow. I agree with your statement in the comment, but I don't see that in the text above.

mingpeiwk replied:
@dthaler How about the following revision?

"also install malicious TAs to potentially many devices under the condition that it can have a Trusted Component signer key that is trusted by the TEEs."

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
+other TAs may not get resource to be installed or properly function. It may
+also install malicious TAs to, potentially, many devices, if it can also 
+get or access a Trusted Component signer key that is trusted by the TEEs. 
+This makes TAMs high value targets. An advanced, persistent threat actor 
```
```suggestion
This makes TAMs high value targets. An advanced persistent threat actor 
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
+also install malicious TAs to, potentially, many devices, if it can also 
+get or access a Trusted Component signer key that is trusted by the TEEs. 
+This makes TAMs high value targets. An advanced, persistent threat actor 
+could potentially compromise a TAM without impacting its certificate or 
+raising concern from the TAM's operator.
```
Not sure what this sentence is trying to say.  In this sense, what is an "advanced persistent threat actor"?

mingpeiwk replied:
I believe Brendon meant to say APT on the TAM system. I just removed it to not say how. Just leave the risk part that "a TAM could be compromised without affecting its certificate".

Please review again. @bremoran @dthaler

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
+could potentially compromise a TAM without impacting its certificate or 
+raising concern from the TAM's operator.
+
+To mitigate this threat, TEE Agents have several options as listed below but 
+not limit to these for detecting and mitigating a compromised TAM:
```
```suggestion
To mitigate this threat, TEEP Agents and Device Owners have several options, including but potentially not limited to those listed below, for detecting and mitigating a compromised TAM:
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
+To mitigate this threat, TEE Agents have several options as listed below but 
+not limit to these for detecting and mitigating a compromised TAM:
 
 1. Apply an ACL to the TAM, indicating which TAs it is permitted to install.
```
```suggestion
1. Apply an ACL to the TAM key, limiting which TAs the TAM is permitted to install.
```

mingpeiwk replied:
Hmm, need clarification on this.

An ACL is to include a list of TA by name or ID such that a remote TAM will not be able to install new TAs, achieving limited damage. Such an ACL may be implemented in REE, for example, applying to the SUIT manifest processing time, which contains TA IDs and TAMs that will be used. In such an implementation, a TAM key isn't accessible. However, it can still limit a compromised TAM's damage via ACL over TAM URL.

An ACL could be implemented and checked inside TEE Agent too in validating a TAM. In this case, TEEP Agent needs to expose an API interface for some REE applications to update TA list in an ACL, which differs from trust anchors that can be preloaded and largely remain static.

dthaler replied:
> An ACL is to include a list of TA by name or ID

Disagree.   As Brendan mentioned, it may be a list of install paths, or a list of trusted component signer keys, or whatever else, it need not be a list of TAs.

> Such an ACL may be implemented in REE

Disagree, it should be implemented in the TEE.  The REE may have no visibility at all, and the device is not even required to have a REE.

> In this case, TEEP Agent needs to expose an API interface for some REE applications to update TA list in an ACL

I don't think the document should suggest that an REE application can have any control over TEE data.  The whole point of the TEE is to defend against anything malicious in the REE.

mingpeiwk replied:
ACL in REE is an option in some cases while it isn't bullet proof but it can be easy to implement and apply by quickly limited TAs in a device with the information from the Untrusted Apps that need TAs.

For the option of ACL in TEE, it is possible but there is the problem who and how it gets updated. We can leave it to TEE providers as it is out of scope for TEEP.

The point is whether we limit the ACL to only TEE, not allowing an option in REE with SUIT information as an input.

mingpeiwk replied:
Removed implication of REE option, and let us move on.

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
+not limit to these for detecting and mitigating a compromised TAM:
 
 1. Apply an ACL to the TAM, indicating which TAs it is permitted to install.
 2. Use a transparency log to expose a TA compromise: TAMs publish an 
 out-of-band record of TA releases, allowing a TEE to cross-check the TAs
 delivered against the TA releases in order to detect a TAM compromise.
```
```suggestion
2. Use a transparency log to expose a TAM compromise: TAMs publish an 
out-of-band record of Trusted Component releases, allowing a TEE to cross-check the Trusted Components
delivered against the Trusted Component installs in order to detect a TAM compromise.
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
+
+There are also threats of hostile or abusive TAMs that act out of expectation 
+of Device Administrators, for example, pushing out TAs that contain 
+some data collection or use users' device resources for distributed jobs for 
```
```suggestion
some data collection or use device resources for distributed jobs for 
```
not just user devices

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
+of Device Administrators, for example, pushing out TAs that contain 
+some data collection or use users' device resources for distributed jobs for 
+a TAM. All the potential damanges that a compromised TAM can make can be done by a
+hostile TAM. A hostile TAM differs in knowingly launch some abusive behavior.
```
```suggestion
hostile TAM. A hostile TAM differs in intentionally conducting some abusive behavior.
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
+some data collection or use users' device resources for distributed jobs for 
+a TAM. All the potential damanges that a compromised TAM can make can be done by a
+hostile TAM. A hostile TAM differs in knowingly launch some abusive behavior.
+
```
The main case above is if someone can tamper with the trust anchor store, whether locally or via a compromised TAM inserting a hostile TAM's key.   The mitigation there are trust anchor store protection as discussed in section 12.4, and some secure auditing of changes to the trust anchor store.
(Maybe add the above in text.)

mingpeiwk replied:
The tampering of Trust Anchor store brings a different use case. It is a source of injecting a new hostile TAM rather than an existing one that changed behavior. Will mention this.

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
+a TAM. All the potential damanges that a compromised TAM can make can be done by a
+hostile TAM. A hostile TAM differs in knowingly launch some abusive behavior.
+
+A Device Administrator may find out that the TAM it previously trusted turns out
+to be not trustworthy anymore. In this case, the Device Administrator can remove
+those TAs managed by that TAM from its devices, and stop trusting the TAM by
+changing its TA's TAM. It can switch to use another TAM that can deliver legitimate
+TAs it needs. In some cases, a Device Administrator may want to remove the 
+Trust Anchor entry in its devices that specifically gives trust to the hostile TAM.
```
This paragraph is really about a compromised TAM and should be there not under hostile tam.

mingpeiwk replied:
I considered it differently - a TAM changes to be hostile over time. It was good when it was first chosen by a Device Administrator, and over time the TAM abuses its privilege to sneak out abusive TAs. A compromised TAM may still become trustworthy once it fixes it vulnerability, and a compromised TAM may proactively clean and fix the issue.

An analogy is the social networking companies. Some company starts to collect user PII data for non-authorized use. They are not compromised, and they were trustworthy when a user signs up at the first place but find not trustworthy when the company changes behavior in privacy protection or abuse its monopoly power. The word "hostile" is a little bit strong for such a service provider but it is considered under this category.

Is the word "trustworthy" confusing here? I could mean that a TAM cannot be trusted anymore because it is compromised at the moment, or it cannot be trusted because it intentionally acts beyond the good deed.

dthaler replied:
My view is that TAM that changes over time to be hostile has been compromised... e.g., by an internal employee or management.

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,47 @@ may need to be updated.
+TAs it needs. In some cases, a Device Administrator may want to remove the 
+Trust Anchor entry in its devices that specifically gives trust to the hostile TAM.
+
+There could be a remedy scenario where Device Administrators work with a TAM that 
+voilated the good deed but changed to correct its behavior to restore the trust.
+In such cases, the TAs may be uninstalled and updated by the TAM proactively
+or upon the Device Administrator's request.
```
I don't think this adds anything beyond the previous paragraph.

mingpeiwk replied:
OK, will remove this paragraph.

mingpeiwk committed:
Update draft-ietf-teep-architecture.md

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Update draft-ietf-teep-architecture.md

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
PR feedback

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Reviewer feedback

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Reviewer feedback

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Reviewer feedback

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Reviewer feedback

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Update draft-ietf-teep-architecture.md

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Editorial update

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Editorial update

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Updated TAM threat section

Removed APT wording in compromised TAM section.

Added the threat of compromised trust anchor case that can plant in a new hostile TAM.

mingpeiwk committed:
Revise on compromised TAM impact in installing TAs

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,43 @@ may need to be updated.
+A Device Administrator may find out that the TAM it selected starts to become hostile
+or abusive. In this case, the Device Administrator can remove
+those TAs managed by that TAM from its devices, and stop trusting the TAM by
+changing its TA's TAM. It can switch to use another TAM that can deliver legitimate
```
```suggestion
updating the TEEP Agent's trust anchor store to use another TAM that can deliver legitimate
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,43 @@ may need to be updated.
+or abusive. In this case, the Device Administrator can remove
+those TAs managed by that TAM from its devices, and stop trusting the TAM by
+changing its TA's TAM. It can switch to use another TAM that can deliver legitimate
+TAs it needs. In some cases, a Device Administrator may want to remove the 
+Trust Anchor entry in its devices that specifically gives trust to the hostile TAM.
```
```suggestion
TAs it needs.
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,43 @@ may need to be updated.
+TAs it needs. In some cases, a Device Administrator may want to remove the 
+Trust Anchor entry in its devices that specifically gives trust to the hostile TAM.
+
+A new hostile TAM may be introduced if the Trust Anchor in a TEEP Agent is compromised.
```
```suggestion
A new hostile TAM may be introduced if the Trust Anchor store in a TEEP Agent is compromised.
```

mingpeiwk committed:
Update draft-ietf-teep-architecture.md

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Update draft-ietf-teep-architecture.md

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Update draft-ietf-teep-architecture.md

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Specify ACL to TAM key in TEE

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,42 @@ may need to be updated.
+To mitigate this threat, TEEP Agents and Device Owners have several options, 
+including but potentially not limited to those listed below, for detecting and mitigating a compromised TAM:
+
+1. Apply an ACL to the TAM key, indicating which TAs the TAM is permitted to install.
```
```suggestion
1. Apply an ACL to the TAM key, limiting which Trusted Components the TAM is permitted to install or update.
```

"indicating" makes it sound like the ACL has TA IDs which it need not.

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,23 +1329,42 @@ may need to be updated.
-still a threat present in allowing a potentially compromised TAM to install
-any TA it wants; for example, a known-compromised TA or an explicitly
-malicious TA.
+## Hostile TAM
+
+There are also threats of hostile or abusive TAMs that act out of expectation 
+of Device Administrators, for example, pushing out TAs that contain 
+some data collection or use device resources for distributed jobs for 
+a TAM. All the potential damages that a compromised TAM can make can be done by a
+hostile TAM. A hostile TAM differs in intentionally conducting some abusive behavior.
+
+A Device Administrator may find out that the TAM it selected starts to become hostile
+or abusive. In this case, the Device Administrator can remove
+those TAs managed by that TAM from its devices, and stop trusting the TAM by
+updating the TEEP Agent's trust anchor store to use another TAM that can deliver legitimate
+TAs it needs.
+
+A new hostile TAM may be introduced if the Trust Anchor store in a TEEP Agent is compromised.
+See Section {{#trust-anchor-compromise}} above for Trust Anchor protection.
```
As noted in email, I would prefer not adding this section.

mingpeiwk replied:
Will remove it, and just leave the expanded Compromised TAM to cover both cases. We will respond Ben for this reasoning.

mingpeiwk committed:
Update draft-ietf-teep-architecture.md

Co-authored-by: Dave Thaler <dthaler@microsoft.com>

mingpeiwk committed:
Decide to not add a separate hostile TAM section

Remove it and treat it under Compromised TAM.

mingpeiwk commented:
Change the PR subject to reflect the change in Compromised TAM.

mingpeiwk committed:
Acknowledgements updates

Added Brendan for his help on Compromised TAM.

mingpeiwk commented:
@dthaler @hannestschofenig please review the latest as the final change candidate. Thanks.

hannestschofenig commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,24 +1329,26 @@ may need to be updated.
-2. Use a transparency log to expose a TA compromise: TAMs publish an 
-out-of-band record of TA releases, allowing a TEE to cross-check the TAs
-delivered against the TA releases in order to detect a TAM compromise.
+to determine that the TAM is trustworthy. A compromised TAM may bring multiple threats 
```
```suggestion
. A compromised TAM may bring multiple threats 
```

hannestschofenig replied:
I removed the second part of the sentence since the subsequent text explains why just verifying the TAM certificate is not enough for establishing trustworthiness

hannestschofenig commented:
Left just a small note; the rest of the text is fine

dthaler reviewed (approved):
Ok once Hannes's change is accepted

dthaler committed:
Update draft-ietf-teep-architecture.md

Co-authored-by: Hannes Tschofenig <hannes.tschofenig@gmx.net>

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,24 +1329,26 @@ may need to be updated.
 ## Compromised TAM
 
 Device TEEs are responsible for validating the supplied TAM certificates
```
```suggestion
Device TEEs are responsible for validating the supplied TAM certificates.
```

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -1329,24 +1329,26 @@ may need to be updated.
-2. Use a transparency log to expose a TA compromise: TAMs publish an 
-out-of-band record of TA releases, allowing a TEE to cross-check the TAs
-delivered against the TA releases in order to detect a TAM compromise.
+. A compromised TAM may bring multiple threats 
```
```suggestion
A compromised TAM may bring multiple threats 
```

dthaler committed:
nit: Remove space before period

bremoran commented:
Thanks everyone, I think that this progressed quite well. I do notice that we're no longer addressing an actively hostile TAM. Do you think that those are equivalent? As in my previous discussion of mitigations, some variations of transparency logs will not work against a hostile TAM.

dthaler commented:
> Thanks everyone, I think that this progressed quite well. I do notice that we're no longer addressing an actively hostile TAM. Do you think that those are equivalent? 

Yes I think they are equivalent.

> As in my previous discussion of mitigations, some variations of transparency logs will not work against a hostile TAM.

Nor would some variations work against a compromised TAM.  I think they are equivalent.
